### ISSUE TITLE
New Pandas Version causes `TypeError`, and `InvalidIndexError` on data import

### ISSUE THREAD

**Issue by Japhiolite** [cgre-aachen/gempy#660]
**Describe the bug**
When initiating a gempy model, loading interface and/or orientation data causes a `TypeError` and `InvalidIndexError`. The error message is pasted below. Error occurs, when initiating a model with loading data:

```python
geo_model = gp.create_model('Simple_model')
gp.init_data(geo_model, [0., 10., 0., 2., 0., 10.], [100, 3, 100],
            path_i = '../data/00_1Layer_simple_model_interfaces.csv',
            path_o = '../data/00_1Layer_simple_model_orientation.csv');
```
Relevant `csv` files from this example are here: [interfaces](https://github.com/Japhiolite/gempy-tests/blob/master/data/00_1Layer_simple_model_interfaces.csv), [orientations](https://github.com/Japhiolite/gempy-tests/blob/master/data/00_1Layer_simple_model_orientation.csv). 

Error was caused in a fresh environment using [`pandas 1.4.0`](https://pandas.pydata.org/docs/whatsnew/v1.4.0.html). 

```python
TypeError                                 Traceback (most recent call last)
File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\indexes\base.py:3621, in Index.get_loc(self, key, method, tolerance)
   3620 try:
-> 3621     return self._engine.get_loc(casted_key)
   3622 except KeyError as err:

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\_libs\index.pyx:136, in pandas._libs.index.IndexEngine.get_loc()

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\_libs\index.pyx:142, in pandas._libs.index.IndexEngine.get_loc()

TypeError: 'slice(None, None, None)' is an invalid key

During handling of the above exception, another exception occurred:

InvalidIndexError                         Traceback (most recent call last)
Input In [3], in <module>
----> 1 gp.init_data(geo_model, [0., 10., 0., 2., 0., 10.], [100, 3, 100],
      2             path_i = '../data/00_1Layer_simple_model_interfaces.csv',
      3             path_o = '../data/00_1Layer_simple_model_orientation.csv')

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\gempy\gempy_api.py:185, in init_data(geo_model, extent, resolution, **kwargs)
    182     geo_model.set_regular_grid(extent, resolution)
    184 if 'path_i' in kwargs or 'path_o' in kwargs:
--> 185     read_csv(geo_model, **kwargs)
    187 if 'surface_points_df' in kwargs:
    188     geo_model.set_surface_points(kwargs['surface_points_df'], **kwargs)

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\gempy\gempy_api.py:230, in read_csv(geo_model, path_i, path_o, **kwargs)
    228 if path_i is not None or path_o is not None:
    229     try:
--> 230         geo_model.read_data(path_i, path_o, **kwargs)
    231     except KeyError as e:
    232         raise KeyError('Loading of CSV file failed. Check if you use commas '
    233                        'to separate your data.' + str(e))

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\gempy\core\model.py:1755, in Project.read_data(self, source_i, source_o, add_basement, **kwargs)
   1753 if add_basement is True:
   1754     self._surfaces.add_surface(['basement'])
-> 1755     self.map_stack_to_surfaces({'Basement': 'basement'}, set_series=True)
   1756 self._rescaling.rescale_data()
   1758 self._additional_data.update_structure()

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\gempy\core\model.py:925, in ImplicitCoKriging.map_stack_to_surfaces(self, mapping_object, set_series, sort_geometric_data, remove_unused_series, twofins)
    923 if type(mapping_object) is dict:
    924     series_list = list(mapping_object.keys())
--> 925     self._stack.add_series(series_list)
    926 elif isinstance(mapping_object, pn.Categorical):
    927     series_list = mapping_object['series'].values

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\gempy\core\data_modules\stack.py:315, in Series.add_series(self, series_list, reset_order_series)
    312 self.faults.faults_relations_df.fillna(False, inplace=True)
    314 if reset_order_series is True:
--> 315     self.reset_order_series()
    316 return self

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\gempy\core\data_modules\stack.py:218, in Series.reset_order_series(self)
    214 def reset_order_series(self):
    215     """
    216     Reset the column order series to monotonic ascendant values.
    217     """
--> 218     self.df.at[:, 'order_series'] = pn.RangeIndex(1, self.df.shape[0] + 1)

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\indexing.py:2273, in _AtIndexer.__setitem__(self, key, value)
   2270     self.obj.loc[key] = value
   2271     return
-> 2273 return super().__setitem__(key, value)

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\indexing.py:2228, in _ScalarAccessIndexer.__setitem__(self, key, value)
   2225 if len(key) != self.ndim:
   2226     raise ValueError("Not enough indexers for scalar access (setting)!")
-> 2228 self.obj._set_value(*key, value=value, takeable=self._takeable)

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\frame.py:3871, in DataFrame._set_value(self, index, col, value, takeable)
   3868     return
   3870 series = self._get_item_cache(col)
-> 3871 loc = self.index.get_loc(index)
   3872 dtype = series.dtype
   3873 if isinstance(dtype, np.dtype) and dtype.kind not in ["m", "M"]:
   3874     # otherwise we have EA values, and this check will be done
   3875     #  via setitem_inplace

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\indexes\base.py:3619, in Index.get_loc(self, key, method, tolerance)
   3614 if tolerance is not None:
   3615     raise ValueError(
   3616         "tolerance argument only valid if using pad, "
   3617         "backfill or nearest lookups"
   3618     )
-> 3619 casted_key = self._maybe_cast_indexer(key)
   3620 try:
   3621     return self._engine.get_loc(casted_key)

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\indexes\category.py:471, in CategoricalIndex._maybe_cast_indexer(self, key)
    467 def _maybe_cast_indexer(self, key) -> int:
    468     # GH#41933: we have to do this instead of self._data._validate_scalar
    469     #  because this will correctly get partial-indexing on Interval categories
    470     try:
--> 471         return self._data._unbox_scalar(key)
    472     except KeyError:
    473         if is_valid_na_for_dtype(key, self.categories.dtype):

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\arrays\categorical.py:1956, in Categorical._unbox_scalar(self, key)
   1953 def _unbox_scalar(self, key) -> int:
   1954     # searchsorted is very performance sensitive. By converting codes
   1955     # to same dtype as self.codes, we get much faster performance.
-> 1956     code = self.categories.get_loc(key)
   1957     code = self._ndarray.dtype.type(code)
   1958     return code

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\indexes\base.py:3628, in Index.get_loc(self, key, method, tolerance)
   3623         raise KeyError(key) from err
   3624     except TypeError:
   3625         # If we have a listlike key, _check_indexing_error will raise
   3626         #  InvalidIndexError. Otherwise we fall through and re-raise
   3627         #  the TypeError.
-> 3628         self._check_indexing_error(key)
   3629         raise
   3631 # GH#42269

File ~\Anaconda3\envs\env_gempyP38_test\lib\site-packages\pandas\core\indexes\base.py:5637, in Index._check_indexing_error(self, key)
   5633 def _check_indexing_error(self, key):
   5634     if not is_scalar(key):
   5635         # if key is not a scalar, directly raise an error (the code below
   5636         # would convert to numpy arrays and raise later any way) - GH29926
-> 5637         raise InvalidIndexError(key)

InvalidIndexError: slice(None, None, None)
```
A clear and concise description of what the bug is. Include error messages / 
tracebacks. If useful include screenshots.

**To Reproduce**
Provide detailed steps to reproduce the behavior:

1. Create an environment with gempy using pandas 1.4.0
2. Run the example above with the two example csv linked
...

**Expected behavior**
`gp.init_data` will cause the error pasted above. 

**Screenshots**

**Desktop (please complete the following information):**
  * OS: Windows 10 Pro 
  * GemPy Version 2.2.10
  * Jupyter Version (if applicable)

**Workaround

Freeze `Pandas` to Version 1.3.X (tried with 1.3.4 and it worked) for now in a new GemPy environment.**

**Comment by Japhiolite** [comment]
see also #673

**Comment by Japhiolite** [comment]
for the time being, we decided to fix pandas to version <1.4 for gempy.
Planned refactoring of gempy will likely resolve the issues.
see PR #499
